src/steth/scUSDCv2.sol

```diff
@@ -0,0 +1,593 @@
+    event ProfitSold(uint256 wethSold, uint256 usdcReceived);
+    event EulerRewardsSold(uint256 eulerSold, uint256 usdcReceived);
+
+    // token representing the rewards from the euler protocol
+    ERC20 public constant eulerRewardsToken = ERC20(C.EULER_REWARDS_TOKEN);
+
+    // Uniswap V3 router
+    ISwapRouter public immutable swapRouter;
+
+    // Chainlink pricefeed (USDC -> WETH)
+    AggregatorV3Interface public usdcToEthPriceFeed;
+
+    // Balancer vault for flashloans
+    IVault public immutable balancerVault;
+
+    // mapping of protocol IDs to adapters
+    mapping(uint8 => IAdapter) protocolAdapters;
+
+    // list of supported protocols IDs
+    uint8[] public supportedProtocolIds;
+
+    struct ConstructorParams {
+        address admin;
+        address keeper;
+        ERC4626 scWETH;
+        ERC20 usdc;
+        WETH weth;
+        ISwapRouter uniswapSwapRouter;
+        AggregatorV3Interface chainlinkUsdcToEthPriceFeed;
+        IVault balancerVault;
+    }
+
+    constructor(ConstructorParams memory _params)
+        scUSDCBase(
+            _params.admin,
+            _params.keeper,
+            _params.usdc,
+            _params.weth,
+            _params.scWETH,
+            "Sandclock USDC Vault v2",
+            "scUSDCv2"
+        )
+    {
+        swapRouter = _params.uniswapSwapRouter;
+        usdcToEthPriceFeed = _params.chainlinkUsdcToEthPriceFeed;
+        balancerVault = _params.balancerVault;
+
+        weth.safeApprove(address(swapRouter), type(uint256).max);
+        weth.safeApprove(address(scWETH), type(uint256).max);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            PUBLIC API
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Set the chainlink price feed for USDC -> WETH.
+     * @param _newPriceFeed The new price feed.
+     */
+    function setUsdcToEthPriceFeed(AggregatorV3Interface _newPriceFeed) external {
+        _onlyAdmin();
+
+        if (address(_newPriceFeed) == address(0)) revert PriceFeedZeroAddress();
+
+        usdcToEthPriceFeed = _newPriceFeed;
+    }
+
+    function addAdapter(IAdapter _adapter) external {
+        _onlyAdmin();
+
+        uint8 id = _adapter.id();
+        protocolAdapters[id] = _adapter;
+        supportedProtocolIds.push(id);
+
+        address(_adapter).functionDelegateCall(abi.encodeWithSelector(IAdapter.setApprovals.selector));
+    }
+
+    function removeAdapter(uint8 _protocolId) external {
+        _onlyAdmin();
+        _isSupportedCheck(_protocolId);
+
+        // check if protocol is being used
+        if (protocolAdapters[_protocolId].getCollateral(address(this)) > 0) revert ProtocolInUse(_protocolId);
+
+        delete protocolAdapters[_protocolId];
+
+        // remove from supportedProtocolIds array
+        for (uint256 i = 0; i < supportedProtocolIds.length; i++) {
+            if (supportedProtocolIds[i] == _protocolId) {
```

Comment:
It will be a better idea to use a EnumerableSet.UintSet for the suportedProtocols array. It has a readily available remove method so you won't need this loop. 
<img width="383" alt="Screenshot 2023-05-22 at 8 46 07 PM" src="https://github.com/lindy-labs/sandclock-contracts/assets/47485188/267c276f-c155-4685-99eb-fb259010ea07">

Reply:
i've gone one step ahead and used `EnumerableMap.UintToAddressMap` for `protocolAdapters` mapping, so `supportedProtocolIds` array is removed entirely, check [this](https://github.com/lindy-labs/sandclock-contracts/pull/58/commits/4d65e879d514898d63934a888558340bf3fbd59c) commit

Reply:
nice one